src/core_atmosphere/physics/physics_mmm/bl_mynn_subroutines.F

```diff
@@ -0,0 +1,6565 @@
+
+       IF ( q3sq .LT. q2sq ) THEN
+          !Apply Helfand & Labraga mod
+          qdiv = SQRT( q3sq/q2sq )   !HL89: (1-alfa)
+!
+          !Use level 2.5 stability functions
+          !e1   = q3sq - e1c*ghel*a2fac
+          !e2   = q3sq - e2c*ghel*a2fac
+          !e3   = e1   + e3c*ghel*a2fac**2
+          !e4   = e1   - e4c*ghel*a2fac
+          !eden = e2*e4 + e3*e5c*gmel
+          !eden = MAX( eden, 1.0d-20 )
+          !sm(k) = q3sq*a1*( e3-3.0*c1*e4       )/eden
+          !!JOE-Canuto/Kitamura mod
+          !!sh(k) = q3sq*a2*( e2+3.0*c1*e5c*gmel )/eden
+          !sh(k) = q3sq*(a2*a2fac)*( e2+3.0*c1*e5c*gmel )/eden
+          !sm(k) = Prnum*sh(k)
+          !sm(k) = sm(k) * qdiv
+
+          !Use level 2.0 functions as in original MYNN
+          sh(k) = sh(k) * qdiv
+          sm(k) = sm(k) * qdiv
+        !  !sm_pbl = sm(k) * qdiv
+        !
+        !  !Or, use the simple Pr relationship
+        !  sm(k) = Prnum*sh(k)
+        !
+        !  !or blend them:
+        !  zi2   = MAX(zi, 300.)
+        !  wt    =.5*TANH((zw(k) - zi2)/200.) + .5
+        !  sm(k) = sm_pbl*(1.-wt) + sm(k)*wt
+
+          !Recalculate terms for later use
+          !JOE-Canuto/Kitamura mod
+          !e1   = q3sq - e1c*ghel * qdiv**2
+          !e2   = q3sq - e2c*ghel * qdiv**2
+          !e3   = e1   + e3c*ghel * qdiv**2
+          !e4   = e1   - e4c*ghel * qdiv**2
+          e1   = q3sq - e1c*ghel*a2fac * qdiv**2
+          e2   = q3sq - e2c*ghel*a2fac * qdiv**2
+          e3   = e1   + e3c*ghel*a2fac**2 * qdiv**2
+          e4   = e1   - e4c*ghel*a2fac * qdiv**2
+          eden = e2*e4 + e3*e5c*gmel * qdiv**2
+          eden = MAX( eden, 1.0d-20 )
+          !!JOE-Canuto/Kitamura mod
+          !!sh(k) = q3sq*a2*( e2+3.0*c1*e5c*gmel )/eden  - retro 5
+          !sh(k) = q3sq*(a2*a2fac)*( e2+3.0*c1*e5c*gmel )/eden
+          !sm(k) = Prnum*sh(k)
+       ELSE
+          !JOE-Canuto/Kitamura mod
+          !e1   = q3sq - e1c*ghel
+          !e2   = q3sq - e2c*ghel
+          !e3   = e1   + e3c*ghel
+          !e4   = e1   - e4c*ghel
+          e1   = q3sq - e1c*ghel*a2fac
+          e2   = q3sq - e2c*ghel*a2fac
+          e3   = e1   + e3c*ghel*a2fac**2
+          e4   = e1   - e4c*ghel*a2fac
+          eden = e2*e4 + e3*e5c*gmel
+          eden = MAX( eden, 1.0d-20 )
+
+          qdiv = 1.0
+          !Use level 2.5 stability functions
+          sm(k) = q3sq*a1*( e3-3.0*c1*e4       )/eden
+        !  sm_pbl = q3sq*a1*( e3-3.0*c1*e4       )/eden
+          !!JOE-Canuto/Kitamura mod
+          !!sh(k) = q3sq*a2*( e2+3.0*c1*e5c*gmel )/eden
+          sh(k) = q3sq*(a2*a2fac)*( e2+3.0*c1*e5c*gmel )/eden
+        !  sm(k) = Prnum*sh(k)
+
+        !  !or blend them:
+        !  zi2   = MAX(zi, 300.)
+        !  wt    = .5*TANH((zw(k) - zi2)/200.) + .5
+        !  sm(k) = sm_pbl*(1.-wt) + sm(k)*wt
+       END IF !end Helfand & Labraga check
+
+       !Impose broad limits on Sh and Sm:
+       gmelq    = MAX(gmel/q3sq, 1e-8)
```

Comment:
This line generates the following compilation error (with the `gnu` build target):
```
bl_mynn_subroutines.F:1815:33:

 1815 |        gmelq    = MAX(gmel/q3sq, 1e-8)
      |                                 1
Error: GNU Extension: Different type kinds at (1)
make[5]: *** [Makefile:53: bl_mynn_subroutines.o] Error 1
```
One possible resolution would be to change `1e-8` to `1d-8`.